src/ledger/fuzz.zig

```diff
@@ -0,0 +1,330 @@
+            .{},
+            dbGet,
+            .{ &db, &random, &total_action_count, maybe_max_actions },
+        );
+        defer db_get_thread.join();
+
+        var db_count_thread = try std.Thread.spawn(
+            .{},
+            dbCount,
+            .{ &db, &total_action_count, maybe_max_actions },
+        );
+        defer db_count_thread.join();
+
+        var db_contains_thread = try std.Thread.spawn(
+            .{},
+            dbContains,
+            .{ &db, &random, &total_action_count, maybe_max_actions },
+        );
+        defer db_contains_thread.join();
+
+        // Batch API
+        var batch_delete_range_thread = try std.Thread.spawn(
+            .{},
+            batchDeleteRange,
+            .{ &db, &random, &total_action_count, maybe_max_actions },
+        );
+        defer batch_delete_range_thread.join();
+    }
+}
+
+fn performDbAction(
+    action_name: []const u8,
+    comptime func: anytype,
+    args: anytype,
+    count: *std.atomic.Value(u64),
+    max_actions: ?usize,
+) !void {
+    var last_print_msg_count: u64 = 0;
+
+    while (true) {
+        if (max_actions) |max| {
+            if (count.load(.monotonic) >= max) {
+                std.debug.print("{s} reached max actions: {}\n", .{ action_name, max });
+                break;
+            }
+        }
+
+        _ = try @call(.auto, func, args);
+        const current_count = count.load(.monotonic);
+        if ((current_count - last_print_msg_count) >= 1_000) {
+            std.debug.print("{d} {s} actions\n", .{ current_count, action_name });
+            last_print_msg_count = current_count;
+        }
+
+        _ = count.fetchAdd(1, .monotonic);
+    }
+}
+
+fn dbPut(
+    db: *BlockstoreDB,
+    random: *const std.rand.Random,
+    count: *std.atomic.Value(u64),
+    max_actions: ?usize,
+) !void {
+    const key = random.int(u32);
+    var buffer: [61]u8 = undefined;
+    // Fill the buffer with random bytes
+    for (0..buffer.len) |i| {
+        buffer[i] = @intCast(random.int(u8));
+    }
+    const value: []const u8 = buffer[0..];
+    try performDbAction(
+        "RocksDb.put",
+        BlockstoreDB.put,
+        .{ db, cf1, (key + 1), Data{ .value = value } },
+        count,
+        max_actions,
+    );
+}
+
+fn dbDelete(
+    db: *BlockstoreDB,
+    random: *const std.rand.Random,
+    count: *std.atomic.Value(u64),
+    max_actions: ?usize,
+) !void {
+    const key = random.int(u32);
+    try performDbAction(
+        "RocksDb.delete",
+        BlockstoreDB.delete,
+        .{ db, cf1, key },
+        count,
+        max_actions,
+    );
+}
+
+fn dbDeleteFilesInRange(
+    db: *BlockstoreDB,
+    random: *const std.rand.Random,
+    count: *std.atomic.Value(u64),
+    max_actions: ?usize,
+) !void {
+    const start = random.int(u32);
+    const end = blk: {
+        const end_ = random.int(u32);
+        if (end_ < start)
+            break :blk (end_ +| start)
+        else
+            break :blk end_;
+    };
```

Comment:
since the values are random -- i think this is the same but simpler
```suggestion
    const start = random.int(u32);
    const end = random.int(u32) +| start;
```

Reply:
Applied as part of the commit that checks the methods return expected output

https://github.com/Syndica/sig/pull/452/commits/9e06b51485f7c019d6d2404e4ade42e270ac008a